_source/_posts/2022-05-16-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,733 @@
+Go to **Account** and **Sign in**. You should be directed to the Okta sign-in form. 
+
+{% img blog/jhipster-k8s-azure/signing-in.png alt:"Okta Login" width:"500" %}{: .center-image }
+
+
+You should be able to authenticate with your Okta credentials.
+
+{% img blog/jhipster-k8s-azure/authenticated.png alt:"Authenticated" width:"800" %}{: .center-image }
+
+Make sure all parts of the application are working. First, test the store (which is using the Cosmos DB Mongo database) by going to **Entities** and **Product**. Make sure you can add a new product and see it in the list of products. Next, create a blog (**Entities** and **Blog**).
+
+If that works, you can take a look at the **Administration** menu. There's a lot of helpful info there. 
+
+You can also check the JHipster Registry: http://localhost:8761/
+
+{% img blog/jhipster-k8s-azure/registry.png alt:"Authenticated" width:"800" %}{: .center-image }
+
+
+## Encrypt the client secret
+
+Leaving secrets in plain text in repositories is a security risk. There are two values in this app that are sensitive: the Cosmos DB connection string that includes the username and password and the Okta OIDC app client secret. You were able to avoid exposing the database credentials by using a `.env` file. However, the Okta client secret is exposed as plain text in the Spring Cloud Config file (`docker-compose/central-server-config/application.yml`). This can be encrypted using the JHipster registry.
+
+Open the registry (http://localhost:8761/) and click on **Configuration** and **Encryption**.
+
+{% img blog/jhipster-k8s-azure/encryption.png alt:"Authenticated" width:"800" %}{: .center-image }
+
+Paste your client secret in the text box. Click **Encrypt**.
+
+This, incidentally, is why you needed the `ENCRYPT_KEY` property in the `.env` file.  That's the key that JHipster registry uses to encrypt these values (so keep it secret!).
+
+You can now copy the encrypted value and paste it back into the `application.yml` file. Make sure you include the `{cipher}` part. It should look similar to below. Don't forget the quotes!
+
+`docker-compose/central-server-config/application.yml`
+
+```yaml
+...
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: https://dev-123456.okta.com/oauth2/default
+        registration:
+          oidc:
+            client-id: 0oa6ycm987987uy98
+            client-secret: "{cipher}88acb434dd088acb434dd088acb434dd0..."
+```
+
+Stop the app if you have not already using `control-c`. Restart it.
+
+```bash
+docker-compose up
+```
+
+Make sure you can still sign into the gateway: http://localhost:8080/
+
+You may be wondering (like I did initially), why can't I just put the client secret in the `.env` file? This doesn't work because the `.env` file is processed by Spring Cloud Config and JHipster registry after the container is composed, not Docker Compose during the container creation, which is when the `.env` file is processed.
```

Comment:
```suggestion
You may be wondering (like I did initially), why can't I just put the client secret in the `.env` file? This doesn't work because the `.env` file is processed by Spring Cloud Config and JHipster registry after the container is composed, not by Docker Compose during the container creation, which is when the `.env` file is processed.
```
clarity